### ISSUE TITLE
Suggest to remove unneeded trailing comma

### ISSUE THREAD

**Issue by nyurik** [rust-lang/rust-clippy#13965]
### What it does

In most cases, a trailing comma before parentheses is not needed, especially in a macro invocation. Yet, I suspect this cannot be fully automated by `rustfmt` because there are some edge cases which do require that comma to be present.  I think there should be a lint offering to remove it.

### Advantage

- Cleaner, easier to read code

### Drawbacks

extremely rare edge cases might be tricky to detect

### Example

```rust
debug!("Foo={}", 1,);
```

Could be written as:

```rust
debug!("Foo={}", 1);
```

**Comment by sivchari** [comment]
The syntax of calling function is formatted by rustfmt. Presumably,  it's good to check only macro calling.

**Comment by nyurik** [comment]
I agree - this probably should only apply to the macros. In the help text, you may also want to add a comment that a macro declaration should have `$(,)?` at the end to allow optional trailing comma - which would also educate users how to write macro_rules

**Comment by samueltardieu** [comment]
If implemented, I think it should be a restriction lint (which is "allow by default"), because some people prefer having a trailing comma when macro calls (*e.g.,* `vec![]`, or other initializers) span multiple lines.

**Comment by nyurik** [comment]
I think this lint should only apply to a single line macros
